client/state/utils/README.md

```diff
@@ -0,0 +1,18 @@
+Utils
+========
+
+A module for those little helpers that need to interact woth the store.
+
+# Actions
+Used in combination with the Redux store instance `dispatch` function, actions can be used in manipulating the current global state.
+
+## uniqueId( prefix )
+Meant as a replacement for _ and lodash "uniqueId" function.
+Returns unique numeric prefixed by prefix and increments internal state, so the next uniqueId will be different.
```

Comment:
@rralian 
> dispatch() method here which expects an action object. 

`redux-thunk` middleware introduces possibility to pass functions to dispatch. This is the whole source:
```
function thunkMiddleware({ dispatch, getState }) {
  return next => action =>
    typeof action === 'function' ?
      action(dispatch, getState) :
      next(action);
}
```
As you see, if action is a function, it gets called with `dispatch` and `getState`.

I completely agree that mechanics of this may seem funky, but I kinda feel that this is the most convenient version to use later.
On the other hand, I recently argued that readability is more important then convenience, and it seems that @dmsnell rightly pointed that readability is in the eye of the reader :)

More to the point: 

> the output of action( dispatch, getState ) — which is the invocation of the thunk — is never caught.

I let myself translate this source to ES5, and as I understand it, it gets returned:
```
function thunkMiddleware(_ref) {
  var dispatch = _ref.dispatch;
  var getState = _ref.getState;

  return function (next) {
    return function (action) {
      return typeof action === 'function' ? action(dispatch, getState) : next(action);
    };
  };
}
```

@mcsf:
> by taking this shortcut — vs. going the regular Redux approach of accessing global state via connect+selector

I took this approach to have an 'atomic' action that both increments value and returns it.
The regular redux approach would be to have action for incrementing and selector for accessing, so to get the next id, the flow would be:
1. Call the action, store value gets incremented
2. Component gets re-rendered because we connected to the store for accessing this data
3. Get new id from the props
4. Use the new id
5. call the action that needs ID
6. Store mutation, re-render...